promql/parser/ast.go

```diff
@@ -119,12 +119,24 @@ type Call struct {
 	PosRange posrange.PositionRange
 }
 
+type MatrixSelectorBinOp struct {
+	Op  ItemType
+	Num *NumberLiteral
+}
+
+type MatrixSelectorBinOps struct {
+	Ops []*MatrixSelectorBinOp
+
+	PosRange posrange.PositionRange
+}
+
 // MatrixSelector represents a Matrix selection.
 type MatrixSelector struct {
 	// It is safe to assume that this is an VectorSelector
 	// if the parser hasn't returned an error.
 	VectorSelector Expr
 	Range          time.Duration
+	BinaryOps      *MatrixSelectorBinOps
```

Comment:
Maybe we can just have regular old BinOp here and assume that parser rejects anything where RHS of the root node is not recursively a number literal?